guppylang/tys/parsing.py

```diff
@@ -122,15 +133,27 @@ def arg_from_ast(
 _type_param = TypeParam(0, "T", True)
 
 
+def type_with_flags_from_ast(
+    node: AstNode,
+    globals: Globals,
+    param_var_mapping: dict[str, Parameter] | None = None,
+) -> tuple[Type, InputFlags]:
+    """Turns an AST expression into a Guppy type possibly annotated with @flags."""
+    # Parse an argument and check that it's valid for a `TypeParam`
```

Comment:
Yes, it's amusing to think that we could have got this far with `x: 6 @inout` ;-)